lib/hash.js

```diff
@@ -1,8 +1,24 @@
 'use strict'
 
+function getInvalidatingOptions (config) {
+  return [
+    'instrument',
```

Comment:
I have to question - what is gained by allowing nyc to perform caching if `instrument` is disabled?  I believe babel has it's own cache so maybe caching should be disabled if `!this.config.caching || !this.config.instrument || this.config.instrumenter === 'noop'`?  Just seems like it might be wasteful at best, damaging at worst.

CC @bcoe @JaKXz thoughts?

Reply:
@taye for now lets not worry about modifying the logic for disabling caching, I'll investigate that later.  Please address my other comments and we can move forward with this.